CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+# Contributing to API Platform
+
+First of all, thank you for contributing, you're awesome!
+
+To have your code integrated in the API Platform project, there is some rules to follow, but don't panic, it's easy!
+
+## Reporting bugs
+
+If you happen to find a bug, we kindly request you to report it. However, before submitting it, please:
+
+  * Check the [project documentation available online](https://api-platform.com/doc/)
+
+Then, if it appears that it's a real bug, you may report it using Github by following these 3 points:
+
+  * Check if the bug is not already reported!
+  * A clear title to resume the issue
+  * A description of the workflow needed to reproduce the bug,
+
+> _NOTE:_ Don’t hesitate giving as much information as you can (OS, PHP version extensions...)
+
+## Pull requests
+
+### Writing a Pull Request
+
+First of all, you must decide on what branch your changes will be based. If the changes your are going to make are
+fully backward-compatible, you should base your changes on the latest stable branch (`1.0` at the moment).
+Otherwise, you should base your changes on the `master` branch.
+
+### Matching coding standards
+
+The API Platform project follows [Symfony coding standards](https://symfony.com/doc/current/contributing/code/standards.html).
+But don't worry, you can fix CS issues automatically using the [PHP CS Fixer](http://cs.sensiolabs.org/) tool
+
+```bash
+php-cs-fixer.phar fix
+```
+
+And then, add fixed file to your commit before push.
+Be sure to add only **your modified files**. If another files are fixed by cs tools, just revert it before commit.
+
+### Sending a Pull Request
+
+When you send a PR, just make sure that:
+
+* You add valid test cases (Behat and PHPUnit).
```

Comment:
What about unit tests vs functional tests ? For example, using [KernelTestCase](https://github.com/symfony/framework-bundle/blob/master/Test/KernelTestCase.php) instead of mocking related objects ?